src/migrations.js

```diff
@@ -0,0 +1,31 @@
+import { produce, setAutoFreeze } from 'immer';
+
+export const migrate = (
+  data,
+  migrations,
+) => {
+  setAutoFreeze(false);
+
+  let version = data._migrationVersion ?? 0;
```

Comment:
Q: Should we start migration versions from `0` or `1`? Meaning, a user adds their first migration:

```ts
migrations: {
  migrationVersion: 0,
  
  0: (state) => { 
    ...
  }
}
```
Or...

```ts
migrations: {
  migrationVersion: 1,
  
  1: (state) => { 
    ...
  }
}
```

Since we're not using an array here I rather like the latter option, but perhaps it could be confusing. Thoughts?

Reply:
Likewise

Reply:
Cool, will resolve